packages/commonwealth/server/eventHandlers/rabbitMQ.ts

```diff
@@ -1,13 +1,14 @@
 import { CWEvent, IEventHandler } from 'chain-events/src';
```

Comment:
Also, maybe this is a JS convention I'm not aware of, but why would we name a class with `I<TheThing`? Normally when I see that I I think of a TS Type for the variable of the same name, but maybe I'm missing something.

Reply:
IEventHandler is the interface implemented by event handlers. It's an old Typescript pattern that we might want to start getting away from.

Reply:
The I<Thing> I'm not entirely sure, this predates me somewhat but Ig it's cause it's an abstract class similar to an interface.

IEventHandler is inside chain-events/src because all the code for it is chain-events specific i.e. it is built using types from the chain-events library and is relevant to chain-events/chain-events npm package. Removing it from chain-events to common would mean chain-events library would not have a definition for what an event handler should look like. This is a remanent of the days when chain-events was just a library.

Reply:
but in this case it's a class, I think for interfaces it makes sense?
```
// handles individual events by sending them off to storage/notifying
export abstract class IEventHandler<DBEventType = IChainEventData> {
  // throws on error, returns a db event, or void
  public abstract handle(
    event: CWEvent,
    dbEvent?: DBEventType
  ): Promise<DBEventType>;
}
```

Reply:
aha I see abstract class kinda Interface. Got it